.pre-commit-config.yaml

```diff
@@ -0,0 +1,16 @@
+repos:
```

Comment:
Do you know what it would take to make `api_gen` a pre-commit too?

It would make sense to have it first.

Reply:
Did this too. The reason I did not do this initially is because it kept throwing an error. Had to look up some things - for example, pre-commit runs scripts concurrently, so had to ensure the `api_gen.sh` script runs serially (otherwise, copying from the repo to the temporary build directory errors out).